gpkit/nomials.py

```diff
@@ -169,10 +169,8 @@ def diff(self, wrt):
         """
         if wrt in self.varkeys:
             wrt = self.varkeys[wrt]
-        elif isinstance(wrt, Monomial):
-            vks = list(wrt.exp)
-            if len(vks) == 1:
-                wrt = vks[0]
+        elif not isinstance(wrt, VarKey):
+            wrt = wrt.varkey
```

Comment:
Heh, good call. Probably don't want people creating varkeys out of Monomials, even if that monomial has only one variable...